fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -303,10 +303,12 @@ def reduce(self) -> None:
         ), "No grads waiting to be reduced, maybe that this was called twice or there was no BW pass ?"
 
         # Trigger all the current BW hooks
-        _ = map(lambda x: x(), self._grad_accs)
+        self._bucket_flush_callback_set = True  # no need to flush in the end, we own the callback execution
+        _ = list(map(lambda x: x(), self._manual_reduce))
```

Comment:
list() was missing to the left, so the execution was always skipped. The unit test was not checking for the grad norm in that case, so this slipped in. Both are fixed (the callbacks are properly executed, and this is properly tested)